### PR TITLE
Set ReferenceAssemblyAttribute in reference assemblies

### PR DESCRIPTION
**Pull request by natemcmaster**
Reference assemblies in .NET are supposed to have these two attributes, which prevent the runtime from loading and executing them.

For other examples of this, see https://github.com/dotnet/corefx/blob/41489a93acf3f36abcaaaea2003a8fdbb577cf35/eng/ReferenceAssemblies.props#L23-L28
and https://github.com/dotnet/standard/blob/14b6385e8640328e33883a919eca11594e22f21c/eng/versioning.targets#L48-L53

### CODE REVIEW

File: src/Shared/src/ReferenceAssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+// All reference assemblies should have the 0x70 flag which prevents them from loading
+// and the ReferenceAssemblyAttribute.
+[assembly:System.Runtime.CompilerServices.ReferenceAssembly]
+[assembly:System.Reflection.AssemblyFlags((System.Reflection.AssemblyNameFlags)0x70)]
```
Comment (davidfowl) [inline src/Shared/src/ReferenceAssemblyInfo.cs]: What flags are these and why can't we use the enum?

Comment (natemcmaster) [reply src/Shared/src/ReferenceAssemblyInfo.cs]: This flag sets AssemblyName.ProcessorArchitecture to None. There is no public API for this. https://github.com/dotnet/coreclr/blob/64ca544ecf55490675e72b853e98ebc8cc75a4fe/src/System.Private.CoreLib/src/System/Reflection/AssemblyName.CoreCLR.cs#L74

Comment (davidfowl) [reply src/Shared/src/ReferenceAssemblyInfo.cs]: Add a comment and link to the source in our source?

Comment (natemcmaster) [reply src/Shared/src/ReferenceAssemblyInfo.cs]: Done

**Pull request merged**